### DISCUSSION TITLE
Please cancel the setting of not installing the recommend package in the installation image

### DISCUSSION THREAD

**Discussion by tcitr-chen** [SpiralLinux/SpiralLinux-project#75] [General]
Please cancel the setting of not installing the recommend package in the installation image, The reason why my fcitx installation and configuration failed.
https://github.com/SpiralLinux/SpiralLinux-project/issues/72
![图片](https://user-images.githubusercontent.com/61141667/183006874-81047d9d-4c82-4f8e-a5e4-a9b88195c408.png)
![图片](https://user-images.githubusercontent.com/61141667/183007004-deaa1651-e40b-417f-90bd-2d8a0c090682.png)

**Comment by geckolinux** [comment]
Hello, thanks for the suggestion. In general it is best to simply improve the documentation so that users know how to install it, as automatic installation of recommended packages leads to a _LOT_ of undesirable package bloat.

**Comment by tcitr-chen** [comment]
Although I can manually remove this setting, I still want to remove this setting on the release image, because this setting will cause many programs to fail to run or error.

**Reply by geckolinux** [reply]
This will not be changed. In practice most things work well even with recommended packages not installed by default. That is why I strongly recommend Synaptic, because you can simply right-click on a package and install the recommended and/or suggested packages if desired.

**Reply by NikkiSatmaka** [reply]
I found that 00recommends, 99norecommends, 99synaptic essentially do the same thing. Is there any reason why there are 3 files for this no recommend config?

I'm quite new to debian and I really like and appreciate what you're doing here.

So is it really quite safe to not install recommended packages?
How should we know whether we should or should not install them?
And so you recommend using Synaptic instead of apt because we can choose the recommended packages one by one?

**Reply by geckolinux** [reply]
@NikkiSatmaka Hi there, thanks for the comment, I'm glad you like the project.

> I found that 00recommends, 99norecommends, 99synaptic essentially do the same thing

Hmm, interesting catch. I add the `99norecommends` file manually to the filesystem, and I suspect that the build tool used to generate the ISO probably added the `00recommends` file automatically. I'm not seeing a `99synaptic` file though.

> So is it really quite safe to not install recommended packages?
> How should we know whether we should or should not install them?

Yep, it's totally safe. And not installing recommended packages by default really does save an incredible amount of bloat. If you install a package and it doesn't work at all or it's missing functionality, then you can assume that it's missing a recommended package, just right-click on it in Synaptic and see if it shows any recommended or suggested packages that aren't installed. In my experience it's rarely an issue though, and the benefits of not installing recommends far outweighs the occasional drawbacks.

**Reply by NikkiSatmaka** [reply]
`99synaptic` is basically the same as `99norecommends`, without the comments
```
$ ls /etc/apt/apt.conf.d
00recommends  01autoremove  01autoremove-kernels  20listchanges  20packagekit  50appstream  60icons  70debconf  80snapper  99forceipv4  99norecommends  99synaptic

$ cat 00recommends
APT::Install-Recommends "false";
Aptitude::Recommends-Important "false";

$ cat 99norecommends
// Don't auto-install 'recommended' packages
// https://wiki.debian.org/AptPreferences
Apt::Install-Recommends "false";%

$ cat 99synaptic
APT::Install-Recommends "false";
``` 
I checked out a couple of recommended packages and yeah, I found most of them are not actually needed. There are few that I actually need though, so I guess this forces me to be more mindful with what I install

**Reply by tcitr-chen** [reply]
But obviously this is an x86 64 system. Faced with the desktop computer, I think it is possible to open and install the Recommend package, but cancel the installation of the suggest package. This is also the choice of most distributions on the market.
If you provide an image of Raspberry Pi or arm architecture, in order to reduce the size, you can choose to close the installation of the Recommend package by default.

**Reply by geckolinux** [reply]
@tcitr-chen The default will not be changed in SpiralLinux, but you are welcome to change it on your personal installation.

**Comment by tcitr-chen** [comment]
There is also a strange setting in the release image that causes problems with case switching, also requesting a cancellation.
![图片](https://user-images.githubusercontent.com/61141667/183084351-f5a6191e-b4fa-4366-8496-41ed42aba2a5.png)
There is no tick in the original debian.

**Reply by geckolinux** [reply]
Correct, this is in intentional decision in SpiralLinux. The Shift key is for case switching, whereas the relatively useless CapsLock key is much more useful for switching to different keyboard layouts.
